tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5788,8 +5905,10 @@ SpirvEmitter::tryToAssignToMatrixElements(const Expr *lhs,
     // If we are writing to only one element, the rhs should already be a
     // scalar value.
     auto *rhsElem = rhs;
-    if (accessor.Count > 1)
-      rhsElem = spvBuilder.createCompositeExtract(elemType, rhs, {i});
+    if (accessor.Count > 1) {
+      rhsElem = spvBuilder.createCompositeExtract(elemType, rhs, {i},
+                                                  lhs->getLocStart());
```

Comment:
should this be `rhs->getLocStart()` ? seems that the code is extracting elements from rhs.